src/org/freedesktop/gstreamer/timecode/GstVideoTimeCode.java

```diff
@@ -0,0 +1,96 @@
+package org.freedesktop.gstreamer.timecode;
+
+import com.sun.jna.Pointer;
+import java.util.stream.Stream;
+import org.freedesktop.gstreamer.MiniObject;
+import org.freedesktop.gstreamer.glib.NativeObject;
+import org.freedesktop.gstreamer.glib.Natives;
+import static org.freedesktop.gstreamer.glib.Natives.registration;
+import static org.freedesktop.gstreamer.lowlevel.GstMetaApi.GstVideoTimeCodeStruct;
+
+/**
+ * A representation of a SMPTE time code.
+ * @author Jokertwo
+ * @see <a href="https://docs.gstreamer.com/documentation/video/gstvideotimecode.html?gi-language=c#GstVideoTimeCode">GstVideoTimeCode</a>
+ */
+public class GstVideoTimeCode extends MiniObject {
```

Comment:
VideoTimeCode. Really a MiniObject? Do we really need a separate package for this?

Reply:
I do not know, how to resolve if is it MiniObject or not?
* [Documentation](https://gstreamer.freedesktop.org/documentation/video/gstvideometa.html?gi-language=c#GstVideoTimeCodeMeta)
* [Source code](https://gitlab.freedesktop.org/gstreamer/gst-plugins-base/-/blob/1.10/gst-libs/gst/video/gstvideotimecode.h#L73)

Reply:
You should see a GstMiniObject as first element of the struct - eg. https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/master/gst/gstbuffer.h#L265  Or you could try calling various bits of the MiniObject API on them, which will likely crash! :smile: Hence need to be careful here.  Likely that you'll want to extend NativeObject directly if anything - check out other classes that directly extend that.